dcicutils/misc_utils.py

```diff
@@ -785,6 +801,27 @@ def check_true(test_value, message, error_class=None):
         raise error_class(message)
 
 
+def remove_element(elem, lst, raise_error=True):
+    """
+    Returns a copy of the given list with the first occurrence of the given element removed.
+
+    If the element doesn't occur in the list, an error is raised unless given raise_error=False,
+    in which case a copy of the original list is returned.
+
+    :param elem: an object
+    :param lst: a list
+    :param raise_error: a boolean (default True)
+    """
+
+    result = lst.copy()
```

Comment:
I would put a comment here saying this is in fact a "true" copy, as this to me is typically read as a shallow copy (I know it isn't).

Reply:
Oh, fascinating. In fact, I thought it a shallow copy. I'll have to investigate that.

Reply:
In fact list.copy() does shallow copying, which had been my expectation and was what I wanted, basically only enough copying to allow me to remove the item without affecting the original structure. I updated the doc string to say so.